src/blueprint/operators/protected.rs

```diff
@@ -10,6 +11,20 @@ pub fn update_protected<'a>(
     TryFold::<(&ConfigModule, &Field, &config::Type, &'a str), FieldDefinition, String>::new(
         |(config, field, type_, _), mut b_field| {
             if field.protected.is_some() || type_.protected.is_some() {
+                let is_input = config.types.iter().any(|(_, _type)| {
+                    !_type.interface
+                        && _type.fields.iter().any(|(_, field)| {
+                            field.args.iter().any(|(_, arg)| {
+                                !scalar::is_scalar(&arg.type_of)
+                                    && config.find_type(&arg.type_of).is_some()
```

Comment:
Input types information is pre computed in a hashmap. Check input_types method on config. Use it check

Reply:
Created a helper on `Type` that uses `input_types` for maximum reusability.

Reply:
As we do not use it anywhere else, and I don't see any other methods on `config.rs` receiving a `Config`, I removed it from there to keep the standard and created the `is_input` method on `protected.rs` file.

Feel free to use whatever version seems most convenient (the non-standard one is implemented on [16296d4](https://github.com/tailcallhq/tailcall/pull/1597/commits/16296d471d19b6f68a33b19962b53f4716ebd342))